src/test/java/org/jenkinsci/plugins/workflow/ArtifactManagerTest.java

```diff
@@ -117,52 +115,77 @@ public static void run(@Nonnull JenkinsRule r, @CheckForNull ArtifactManagerFact
             aa.setDefaultExcludes(false);
             p.getPublishersList().add(aa);
             FreeStyleBuild b = r.buildAndAssertSuccess(p);
-            VirtualFile root = b.getArtifactManager().root();
-            new Verify(agent, root, weirdCharacters).run();
-            if (b.getArtifactManager() instanceof StashManager.StashAwareArtifactManager) {
-                TaskListener listener = StreamTaskListener.fromStderr();
-                Launcher launcher = agent.createLauncher(listener);
-                EnvVars env = agent.toComputer().getEnvironment();
-                env.putAll(agent.toComputer().buildEnvironment(listener));
-                // Make sure we can stash and then unstash within a build:
-                StashManager.stash(b, "stuff", ws, launcher, env, listener, "file", null, false, false);
-                ws.child("file").delete();
-                StashManager.unstash(b, "stuff", ws, launcher, env, listener);
-                assertEquals("content", ws.child("file").readToString());
-                ws.child("file").delete();
-                // Copy stashes and artifacts from one build to a second one:
-                p.getPublishersList().clear();
-                FreeStyleBuild b2 = r.buildAndAssertSuccess(p);
-                ExtensionList.lookupSingleton(StashManager.CopyStashesAndArtifacts.class).copy(b, b2, listener);
-                // Verify the copied stashes:
-                StashManager.unstash(b2, "stuff", ws, launcher, env, listener);
-                assertEquals("content", ws.child("file").readToString());
-                // And the copied artifacts:
-                root = b2.getArtifactManager().root();
+            f.apply(agent, p, b, ws);
+        } finally {
+            if (runningContainer != null) {
+                runningContainer.close();
+            }
+        }
+    }
+
+    /**
+     * @param image use {@link #prepareImage} in a {@link BeforeClass} block
+     */
+    public static void artifactArchive(@Nonnull JenkinsRule r, @CheckForNull ArtifactManagerFactory factory, boolean weirdCharacters, @CheckForNull DockerImage image) throws Exception {
```

Comment:
Javadoc should explain that unlike the `…AndDelete` variant, we assert that the manager ignores requests to delete things. Ditto for `artifactStash`.